### DISCUSSION TITLE
How to clean the hitchmap data?

### DISCUSSION THREAD

**Discussion by tillwenke** [Hitchwiki/hitchwiki#200] [Q&A]
Improving the quality of the spots/ data on https://hitchmap.com is planned for a long time.
Currently spots can be placed anywhere that lead to the current state of duplicate and misplaced spots.

We want to discuss all measures to clean the data one by one in single issues here https://github.com/Hitchwiki/hitchmap-data/labels/cleaning. Measures are tagged see: https://github.com/Hitchwiki/hitchmap-data/labels/cleaning. You are free to add more issues for this purpose.
First clarifying if and second how to implement it. Sometimes we have to parameterize a measure - parameters can be identified by an "x". The data belongs to all of us - changing it is a severe intervention. The more real world hitchhiking experience we can incorporate the better. So I am hoping for your participation and that we will find a common ground.

Thus for each measure there should also be a way to ensure that new data adheres to it.

Giving an example:
- measure: we want that a spot within a highway service station is unique and placed at the center of the station
- how to: get the polygon of the station from https://openstreetmap.org (OSM) check if a spot is insight or 10 m outside (placed slightly off) the polygon and move it to the center of the station
- in production: if a new spot is placed inside a service station (check via OSM) only allow to place it at its center

We will apply the measure to the current dataset of spots and give all resulting changes that are not minor to community members to review and approve.

The code for cleaning resides here https://github.com/Hitchwiki/hitchmap-data/tree/main/cleaning.
To make this more realistic we gave it a first try with the measures described here https://github.com/Hitchwiki/hitchmap-data/blob/main/cleaning/README.md which resulted in https://github.com/Hitchwiki/hitchmap-data/blob/main/cleaning/map/map_Germany.html that you can view in your browser.

You are very welcome to request some analysis on the data or do it yourself to boost our discussion.

A further idea could be to set the goal to possibly integrate our spot data into OSM. This implies that spots are assigned to real world features e.g. the road or gas station you are hitchhiking at. This would increase data quality a lot. On the other hand it would take away simplicity and freedom the current "just-place-your-spot-anywhere" policy provides. We are super happy to discuss this below :)

**Comment by Erik0806** [comment]
In general I would totally agree with cleaning up data.
The question is only where to start and where to stop  Should we also do this for toll stations, parking lots, etc.
And sometimes on bigger gas stations it is really practical to see where other hitchers had luck or not. So I really don't know how much the community would really profit from this

**Reply by tillwenke** [reply]
Hey, I agree that there are debatable cleaning measures.
The measures are collected here https://github.com/Hitchwiki/hitchmap-data/labels/cleaning would be nice if you could leave a little reaction on those that you would (not) support from your hitchhiking experience :)

**Reply by Erik0806** [reply]
done :)
